internal/validators/ottl/filter_spec_validator.go

```diff
@@ -0,0 +1,98 @@
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottldatapoint"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottllog"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlmetric"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlresource"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlscope"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+	telemetryv1alpha1 "github.com/kyma-project/telemetry-manager/apis/telemetry/v1alpha1"
+)
+
+type FilterSpecValidator struct {
+	parserCollection *genericParserCollection
+}
+
+func NewFilterSpecValidator(signalType SignalType) (*FilterSpecValidator, error) {
+	opts := newFilterParserCollectionOpts(signalType)
+
+	parserCollection, err := newGenericParserCollection(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &FilterSpecValidator{parserCollection: parserCollection}, nil
+}
+
+func (v *FilterSpecValidator) Validate(filters []telemetryv1alpha1.FilterSpec) error {
+	const errorMessage = "invalid FilterSpec"
+
+	for _, fs := range filters {
+		if err := v.parserCollection.parseConditions(fs.Conditions); err != nil {
+			return &InvalidOTTLSpecError{Err: fmt.Errorf("%s: %w", errorMessage, err)}
+		}
+	}
+
+	return nil
+}
+
+func newFilterParserCollectionOpts(signalType SignalType) []genericParserCollectionOption {
+	var opts []genericParserCollectionOption
+
+	switch signalType {
+	case SignalTypeTrace:
+		opts = append(opts,
+			// Since context inference is not available in the filter processor yet,
+			// we set the context to span as the minimum required context.
+			// Span event context is not supported.
+			withSpanParser(
+				ottlfuncs.StandardConverters[ottlspan.TransformContext](),
+				ottl.WithConditionConverter(nopConditionConverter[ottlspan.TransformContext]),
+			),
+		)
+	case SignalTypeLog:
+		opts = append(opts,
+			withLogParser(
+				ottlfuncs.StandardConverters[ottllog.TransformContext](),
+				ottl.WithConditionConverter(nopConditionConverter[ottllog.TransformContext]),
+			),
+		)
+	case SignalTypeMetric:
+		opts = append(opts,
+			// Since context inference is not available in the filter processor yet,
+			// we set the context to datapoint as the minimum required context.
+			// That is why metric-context-only functions (like HasAttrKeyOnDatapoint or HasAttrOnDatapoint) are not supported here
+			// and only standard converters are included.
+			withMetricParser(
+				ottlfuncs.StandardConverters[ottlmetric.TransformContext](),
+				ottl.WithConditionConverter(nopConditionConverter[ottlmetric.TransformContext]),
+			),
+			withDataPointParser(
+				ottlfuncs.StandardConverters[ottldatapoint.TransformContext](),
```

Comment:
why not use the set of functions specific to the filter processor?
ottl.CreateFactoryMap(filterprocessor.DefaultDataPointFunctions()...)

Although here they would be the same set of functions, I think it is better to use this API in case functions specific to datapoint are added in the future